### PR TITLE
Glooctl interactive messages, fix broken unit test

### PR DESCRIPTION
**Pull request by kdorosh**
Fixes outdated state in `glooctl` interactive prompts by lazily providing the message to prompter.

Addresses the confusing logging and outdated state mentioned in https://github.com/solo-io/gloo/issues/551, as well as the same outdated state bug in a couple other places.

Also fix a broken interactive cli unit test

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/551
resolves https://github.com/solo-io/gloo/issues/387

### CODE REVIEW

**Comment by kdorosh** [comment]
Not sure why https://github.com/solo-io/gloo/issues/655 got linked here by the bot. That is an unrelated change

File: pkg/cliutil/input.go [hunk 1]
```diff
@@ -21,10 +21,22 @@ func GetYesInput(msg string) (bool, error) {
 	return strings.ToLower(yesAnswer) == "y", nil
 }
 
+func GetStringInputLazy(msgProvider func() string, value *string) error {
```
Comment (kdorosh) [inline pkg/cliutil/input.go]: open to renaming these to `GetStringInputLazyMessage`, thoughts?

Comment (rickducott) [reply pkg/cliutil/input.go]: up to you -- I'm fine either way

File: projects/gloo/cli/pkg/cmd/create/upstream_interactive_test.go [hunk 2]
```diff
@@ -156,14 +156,16 @@ var _ = Describe("Upstream Interactive Mode", func() {
...
+	It("should work for static with hosts", func() {
 		testutil.ExpectInteractive(func(c *testutil.Console) {
 			c.ExpectString(upstreamPrompt)
 			c.SendLine("static")
 			c.ExpectString("Add another host for this upstream (empty to skip)? []")
 			c.SendLine("foo")
-			c.SendLine("") // can not figure out how to advance in this case, some idiosyncrasy with the slice prompt
+			c.ExpectString("Add another host for this upstream (empty to skip)? [foo]")
+			c.SendLine("bar")
+			c.ExpectString("Add another host for this upstream (empty to skip)? [foo bar]")
```
Comment (kdorosh) [inline projects/gloo/cli/pkg/cmd/create/upstream_interactive_test.go]: fixing this test also tests that our console output now properly updates state during interactive cli use!

**Pull request merged**